### PR TITLE
Sam 372 reframe credit outputs

### PR DESCRIPTION
**Pull request by brtietz**
Reformat excess generation outputs in accordance with https://github.com/NREL/SAM/issues/372

- All "Excess generation" variables with units other than kWh have been removed or changed
- Net metering and net billing have separate outputs (will be zero for the other)
- Only "credits applied" are shown in $ values
- Buy all / sell all sets all excess generation outputs to zero
- Fixed bug where final month's credits in "net billing with carryover to next month" were not being applied. Now applied during the final month (default December)

One open question: should net billing's excess generation change to sum positive (sub) hourly values? https://github.com/NREL/SAM/issues/372#issuecomment-725020561

Test changes are due to removal of some SSC vartable outputs, and the additional credits due to the "net billing with carryover to next month" bugfix. All rates except "net billing with carryover to next month" should have the same monthly & annual bills as before. No changes to the SAM repository were required.

### CODE REVIEW

File: test/ssc_test/cmod_utilityrate5_test.cpp [hunk 1]
```diff
@@ -214,14 +214,10 @@ TEST(cmod_utilityrate5_eqns, Test_Residential_net_metering_credits_in_may) {
     EXPECT_NEAR(36.6, cost_with_system, 0.1);
 
     int length;
-    ssc_number_t* excess_dollars = ssc_data_get_array(data, "year1_excess_dollars_earned", &length);
+    ssc_number_t* excess_dollars = ssc_data_get_array(data, "year1_nm_dollars_applied", &length);
     float may_dollars = excess_dollars[credit_month];
```
Comment (sjanzou) [inline test/ssc_test/cmod_utilityrate5_test.cpp:218]: Can you please do a sanity check with a credit_month of 0 and another test with the credit_month 12 or higher?

File: test/ssc_test/cmod_utilityrate5_test.cpp [hunk 2]
```diff
@@ -438,7 +430,7 @@ TEST(cmod_utilityrate5_eqns, Test_Residential_TOU_Rates_net_billing_carryover_ap
 
     ssc_number_t cost_with_system;
     ssc_data_get_number(data, "elec_cost_with_system_year1", &cost_with_system);
-    EXPECT_NEAR(-115.6, cost_with_system, 0.1);
+    EXPECT_NEAR(-173.7, cost_with_system, 0.1);
```
Comment (sjanzou) [inline test/ssc_test/cmod_utilityrate5_test.cpp:434]: @brtietz, do you have a workbook to support these updates? Can you place in the SAM-documentation repo in the UtilityRates folder?

**Review by sjanzou — CHANGES_REQUESTED** [review]
Please add the test below but everything looks good...

File: test/ssc_test/cmod_utilityrate5_test.cpp (hunk 1 above)
Comment (brtietz) [reply test/ssc_test/cmod_utilityrate5_test.cpp:218]: Great catch! 0 had some issues which were fixed in the latest push. 12 was handled by the vartable input checks.

File: test/ssc_test/cmod_utilityrate5_test.cpp (hunk 2 above)
Comment (brtietz) [reply test/ssc_test/cmod_utilityrate5_test.cpp:434]: I don't have a workbook, I was looking at the credit earned categories and making sure that the December values were the sum of the Nov credit and the Dec earned (or whatever is appropriate for another month). This SAM file duplicates the gen and load profiles from the unit tests and allows for more transparency: https://github.com/NREL/SAM-documentation/blob/master/UtilityRates/NetBilling/generic_system_ur_tests.sam Good enough?

**Review by sjanzou — APPROVED** [review]
Looks good!

**Review by cpaulgilman — APPROVED** [review]

**Comment by brtietz** [comment]
Merged in f94e504

**Pull request closed without merging**